src/mlagility/cli/report.py

```diff
@@ -1,521 +1,204 @@
+                report[build_name] = BuildResults()
+
+            # Get model hash from build name
+            report[build_name].model_hash = state.config.build_name.split("_")[-1]
+
+            # Get model hash from build name
+            report[build_name].params = _update_numeric_attribute(
+                state.info.num_parameters, report[build_name].params
             )
 
+            # Extract labels (if any)
+            parsed_labels = labels.load_from_cache(cache_dir, build_name)
+            expected_attr_from_label = {
+                "name": "model_name",
+                "author": "author",
+                "class": "model_class",
+                "task": "task",
+            }
+            for label in expected_attr_from_label.keys():
+                results_attr = expected_attr_from_label[label]
+                if label not in parsed_labels.keys():
+                    report[build_name].__dict__[results_attr] = "-"
+                else:
+                    report[build_name].__dict__[results_attr] = parsed_labels[label][0]
+
+            # Get Groq latency and number of chips
+            groq_estimated_latency = get_estimated_groq_latency(build_name, cache_dir)
+            report[build_name].groq_estimated_latency = _update_numeric_attribute(
+                groq_estimated_latency, report[build_name].groq_estimated_latency
+            )
+            report[build_name].groq_chips_used = _update_numeric_attribute(
+                state.num_chips_used, report[build_name].groq_chips_used
+            )
+
+            # Reloading state after estimating latency
+            state = build.load_state(state_path=model_state_yaml)
+
+            # Get CPU latency
+            x86_output_dir = os.path.join(cache_dir, build_name, "x86_benchmark")
+            x86_stats_file = os.path.join(x86_output_dir, "outputs.json")
+            if os.path.isfile(x86_stats_file):
+                with open(x86_stats_file, encoding="utf-8") as f:
+                    g = json.load(f)
+                report[build_name].x86_latency = g.get("Mean Latency(ms)", {})
+
+            # Get GPU latency
+            nvidia_output_dir = os.path.join(cache_dir, build_name, "nvidia_benchmark")
+            nvidia_stats_file = os.path.join(nvidia_output_dir, "outputs.json")
+            if os.path.isfile(nvidia_stats_file):
+                with open(nvidia_stats_file, encoding="utf-8") as f:
+                    g = json.load(f)
+                report[build_name].nvidia_latency = (
+                    g.get("Total Latency", {}).get("mean ", "-").split()[0]
+                )
+
+    # Populate results spreadsheet
+    with open(report_path, "w", newline="", encoding="utf8") as spreadsheet:
+        writer = csv.writer(spreadsheet)
+        cols = BuildResults().__dict__.keys()
+        writer.writerow(cols)
+        for model in report.keys():
+            writer.writerow([report[model].__dict__[col] for col in cols])
```

Comment:
```suggestion
            writer.writerow([vars(report[model])[col] for col in cols])
```

supposed to avoid using `__dict__`

Reply:
Sounds good